compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
Currently we are creating a new TR::MyClass which inherits from OMR::MyClass or MyClass from a project of your choice for the mapping. One thing that jumps to mind is that MyClass could have platform-specific code, in which case we need to create OMR::X86::MyClass and making sure TR::MyClass is mapped to the right MyClass (for the correct platform) requires extra care. In either case, we will need to putting the correct mapping in ValuePropagation.hpp so consumers of the VP class only needs to include this one file.

@0xdaryl Do you have any thoughts on this?